src/pub.rs

```diff
@@ -105,6 +105,7 @@ impl MultiPeer for PubSocketBackend {
 
 pub struct PubSocket {
     pub(crate) backend: Arc<PubSocketBackend>,
+    binds: Vec<Endpoint>,
```

Comment:
Let me know if the right location for this is in `TSocket`, rather than `TSocketBackend`. The actual `binds()` function should probably be in `Socket` either way, as it ought to be externally visible.